### PR TITLE
Rip out starttls.

### LINKED ISSUE #535: SSL
I'm told that this has been discussed before, but I'm not going to search through over 500 entries to find it, so let's start fresh. Thanks to Pixelz for some of the background information.

According to everything I've read, the following are true:

(1) Eggdrop, if SSL is compiled in, can do SSL connections for botnet (if you listen +port)

(2) Eggdrop, if SSL is compiled in, will attempt to do STARTTLS if the bots, on both ends, are SSL-enabled but a non-SSL port is being used for the connection attempt

(3) Eggdrop, if SSL is compiled in, will *NOT* link bots if no certificate is available when STARTTLS is thrown (as per the config settings)

There is a specific problem as a result: for those on "admin [group-policy] limited windows systems" with WinDrop (which is based on Eggdrop, thus this being reported here, before anyone yells at me -- I'll yell right back), the use of OPENSSL may not be available to a given user in a multi-user environment (or a simple, single-user computer due to whatever) where certificate generation isn't wanted, desired, or possible. Between group policies and administrative lock-outs in various configurations make this a real issue.

Currently, I have to put up an off-line (NO-IRC, remember those?) hub bot just to hub a local cluster of automatically-SSL-compiled-in bots because I can't link them together through LOCALHOST. I don't have the option of getting the "non-SSL" port of WinDrop because the botnet they hook into *requires* SSL connections there (I can't make those consistently connect / hold). In addition there are several 1.6.13, 1.6.19, and 1.6.21 bots in there that can't handle the SSL connections anyway. Duct tape, spit, bailing wire, and magic is holding that part of the botnet together.

Now, why shouldn't I have the right (and control) to link my own LOCALHOST bots without being ***FORCED*** to set up certificates for each one??? Shouldn't I have the right to make that decision for myself?

I've been told that there was a suggestion "floating around" (not found in these 500+ issues) of using a port # of "-3333" where "-" says: DO-NOT-USE-SSL/STARTTLS-NO-MATTER-WHAT (in other words, "stay in plaintext"). It seems that, that suggestion either died or just wasn't supported. In either case: why?

Shouldn't we, as bot owners *of any kind and of any version and of any OS (given the various restrictions possible)* have control of that?

Revive the "-port" discussion.

### PR DESCRIPTION
**Pull request by Cizzle**
Found by:
Patch by: Cizzle
Fixes: #535 

One-line summary: Now only TLS-to-TLS and plain-to-plain botlinks are allowed; although plain links can still be upgraded manually.

Also see #552.

### CODE REVIEW

**Comment by vanosg** [comment]
For documentation purposes, 1.8 link tree:

1.8 | 1.9 | Result
----|-----| --------
Links to ()port | Listens with ()port | &#x1F49A; 1.8 upgrades to TLS with STARTTLS
Links to ()port | Listens with +port | &#x1F49A; Tries plain/STARTTLS, fails, then tries TLS, succeeds
Links to +port | Listens with ()port | &#x1F534; Tries TLS to plain port, Fails
Links to +port | Listens with +port | &#x1F49A; Succeeds
 | |
Listens with ()port | Links to ()port | &#x1F49A; 1.8 sends STARTTLS, upgrades to TLS
Listens with ()port | Links to +port | &#x1F534; Tries TLS to plain port, fails
Listens with +port | Links to ()port | &#x1F534; Timeout (Tries plain to TLS port, fails)
Listens with +port | Links to +port | &#x1F49A; Succeeds

In short, a 1.8 bot will still require TLS for a link to a 1.9 bot

**Comment by vanosg** [comment]
LGTM :+1:

**Review by vanosg — APPROVED** [review]
